cds_ils/importer/documents/api.py

```diff
@@ -0,0 +1,128 @@
+import click
+from elasticsearch_dsl import Q
+from elasticsearch_dsl.query import Query, Bool
+from invenio_app_ils.documents.api import Document
+from invenio_app_ils.documents.search import DocumentSearch
+
+from cds_ils.importer.errors import DocumentImportError
+
+
+def check_search_results(result, search_term, search_term_name,
+                         raise_on_empty=False, raise_on_many=True):
+    """Check number of search results"""
+    hits_total = result.hits.total.value
+
+    if hits_total < 1:
+
+        click.secho(
+            "no document found with {} {}".format(search_term_name,
+                                                  search_term),
+            fg="red",
+        )
+        if raise_on_empty:
+            raise DocumentImportError(
+                "no document found with {} {}".format(search_term_name,
+                                                      search_term)
+            )
+    elif hits_total > 1:
+        click.secho(
+            "found more than one document with {} {}".format(search_term_name,
+                                                             search_term),
+            fg="red",
+        )
+        if raise_on_many:
+            raise DocumentImportError(
+                "found more than one document with {} {}".format(
+                    search_term_name, search_term)
+            )
+    return hits_total
+
+
+def search_documents_by_isbn(isbn):
+    """Find document by ISBN."""
+    search = DocumentSearch().query(
+        "bool",
+        must=[
+            Q("match", identifiers__scheme="ISBN"),
```

Comment:
For all of them:
```suggestion
            Q("term", identifiers__scheme="ISBN"),
```
See my comment below.

Reply:
I've tested them and changed accordingly